### ISSUE TITLE
Support a headless mode

### ISSUE THREAD

**Issue by marcoceppi** [ublue-os/yafti#4]
There's a good chance some downstream image makers will want the features of package installation without any popup. One option would be to have YaftiScreen understand headless vs not headless and do an almost `--assume-yes` for each screen. Another option would be to have YaftiScreens describe a separate workflow for when headless. So TitleScreen would be a noop (or print to log) and PackageScreen would jump immediately to the package install step functionall and output to a log vs the console component.

**Comment by EinoHR** [comment]
@marcoceppi 's comment in https://github.com/ublue-os/startingpoint/issues/37#issuecomment-1520554065

> I'm considering moving this into Yafti repo proper. There's an architectural change which will split Yafti into several modual components.
> ## yafti.core
> 
> This will be a silm, almost abstract base, for the minimum required to process rules, config, and templates/plugins
> ## yafti.cli
> 
> A console only version of Yafti which will add the ability for it to be run from a CLI with a rich terminal experience
> ## yafti.gtk
> 
> The current version of Yafti, but rebased on top of the new yafti.core. Requring libadawaita and gtk
> ## yafti.qt
> 
> A QT-only compatible interface for the yafti.core
> 
> This split should allow yafti to work natively with any (or for CLI, no) desktop environment.

**Comment by EinoHR** [comment]
I totally agree with that vision of splitting up yafti into multiple blocks.
I think `yafti.core` should be everything–config parsing, package installation, state management–neatly packed into a library with an elegant api.

For `yafti.core`, though, I'd recommend (and gladly help with) using [Nim](https://nim-lang.org/) as the programming language.
* Python-ish syntax
* Type safety
* Compiled language (compiles to C, JS, and various other intermediaries)
  * Faster runtime and less runtime errors
* Basically a better Go that looks like Python
* Really good interoperability with C and anything that supports using C libraries, so, basically a library built in Nim can be used in many many other languages with just minimal wrapping
  * [Nimpy](https://github.com/yglukhov/nimpy) can be used to very easily create a Python library with Nim, no GUI rewrites needed
  * Nim is both case_insensitive and underscoreInsensitive meaning a Python dev would be totally able to use it with snake_case, while a dev using the library from some other language could use their preferred style
* Reading YAML: [NimYAML](https://nimyaml.org/)

**Comment by abanna** [comment]
@xynydev  where did you guys end up on this one?

**Comment by xynydev** [comment]
Nowhere. I was just idealistically bousting my favorite programming language, not willing to do any work. But good work on your end @abanna .
